Application/src/main/AndroidManifest.xml

```diff
@@ -32,6 +32,8 @@
 
     <uses-permission android:name="android.permission.BLUETOOTH"/>
     <uses-permission android:name="android.permission.BLUETOOTH_ADMIN"/>
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
```

Comment:
What do you need the fine location for?

Reply:
check the note at the end or this section : https://developer.android.com/guide/topics/connectivity/bluetooth-le.html#permissions

Reply:
Isn't the ACCESS_COARSE_LOCATION sufficient then?